skiros2_common/src/skiros2_common/core/world_element.py

```diff
@@ -150,9 +150,13 @@ def getIdNumber(self):
         """
         @brief Return the element id number as integer
         """
-        if self._id.find('-') < 0:
+        hash_pos = self._id.find('#')
```

Comment:
@matthias-mayr I think it would be just easier to replace find with rfind :)

Reply:
rfind for the "-" or the "#"? Assume the former since the latter does not seem to make any sense to me

However just using rfind for the "-" does not necessarily yield to the expected result.
Example:
A malformed URI like `http://www.inf.ufrgs.br/phi-group/ontologies/cora.owl#Robot` that does not have an ID should lead to a return value of "-1" according to the previous/other code.
That's why I wanted to search explicitly after the hash if there's one

Reply:
Right, what about the following? 
```
hash_pos = self._id.rfind('#')
dash_pos = self._id.rfind("-")
return -1 if dash_pos == -1 or dash_pos < hash_pos else int(self._id.split('-')[1])
 ```